### PR TITLE
Fixes bug 1260988 - Added product and version filters to NewSignature…

### PR DESCRIPTION
**Pull request by adngdb**
… service.

### CODE REVIEW

File: webapp-django/crashstats/tools/forms.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    start_date = forms.DateField(required=False)
+    not_after = forms.DateField(required=False)
+    product = MultipleValueField(required=False)
+    version = MultipleValueField(required=False)
+
+    def clean_product(self):
+        """By default, we want the product to be set to something, it should
+        not be empty. """
+        value = self.cleaned_data['product']
+        value = filter(bool, value)  # Remove all empty values.
```
Comment (peterbe) [inline webapp-django/crashstats/tools/forms.py]: Neat but if you need to write a comment perhaps `value = [x for x in value if x]` is better :)

File: webapp-django/crashstats/tools/forms.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    end_date = forms.DateField(required=False)
+    start_date = forms.DateField(required=False)
+    not_after = forms.DateField(required=False)
+    product = MultipleValueField(required=False)
+    version = MultipleValueField(required=False)
+
+    def clean_product(self):
+        """By default, we want the product to be set to something, it should
+        not be empty. """
```
Comment (peterbe) [inline webapp-django/crashstats/tools/forms.py]: This should not be done in the form. Defaults are best dealt with in the view. E.g.
```python
product = form.cleaned_data['product'] or settings.DEFAULT_PRODUCT
```

**Comment by peterbe** [comment]
r+ if you move the default product thing out of the form.

File: webapp-django/crashstats/tools/forms.py (hunk 1 above)
Comment (adngdb) [inline webapp-django/crashstats/tools/forms.py]: Using ``filter`` is faster, and I'm happy I learned about it, it's quite a convenient built-in!

**Pull request merged**